src/components/TokenEnterAmount.tsx

```diff
@@ -18,21 +18,227 @@ import Touchable from 'src/components/Touchable'
+   *
+   * This field consists of two values: token and local. Both values represent calculated values for the
+   * corresponding amount type. Whenever we change token amount - we  need to recalculate both token and
+   * local amounts. Each object consists of:
+   *   - "amount" - this is the actual value, formatted to the unified format without any groupings
+   *     and with a point as a decimal separator, format example: "1234.5678" (as per "amountRaw" value above)
+   *   - "bignum" - this is a BigNumber representation of the "amount" field. Necessary for easier
+   *     condition checks and various processing things.
+   *   - "readable" - this is a read-only component-friendly value that contains all of the necessary
+   *     formatting, including: grouping, decimals, token symbol/fiat sign, small amounts format. This
+   *     value is only necessary to be passed to TokenEnterAmount component fields as:
+   *       - token.readable -> tokenAmount
+   *       - local.readable -> localAmount
+   */
+  const derived = useMemo(() => {
+    if (amountType === 'token') {
+      const parsedTokenAmount = amountRaw === '' ? null : parseInputAmount(amountRaw)
+
+      const tokenToLocal = convertTokenToLocalAmount({
+        tokenAmount: parsedTokenAmount,
+        tokenInfo,
+        usdToLocalRate,
+      })
+
+      const convertedTokenToLocal =
+        tokenToLocal && tokenToLocal.gt(0) ? tokenToLocal.toFixed(2) : ''
+
+      return {
+        token: {
+          amount: amountRaw,
+          bignum: parsedTokenAmount,
+          readable: getDisplayTokenAmount(parsedTokenAmount, props.token),
+        },
+        local: {
+          amount: convertedTokenToLocal,
+          bignum: tokenToLocal,
+          readable: getDisplayLocalAmount(tokenToLocal, localCurrencySymbol),
+        },
+      }
+    }
+
+    /**
+     * At this point, we can be sure that we are processing local (fiat) input.
+     */
+    const parsedLocalAmount = amountRaw === '' ? null : parseInputAmount(amountRaw)
+
+    const localToToken = convertLocalToTokenAmount({
+      localAmount: parsedLocalAmount,
+      tokenInfo,
+      usdToLocalRate,
+    })
+
+    const convertedLocalToToken =
+      localToToken && localToToken.gt(0)
+        ? // no group separator for token amount, round to token.decimals and strip trailing zeros
+          localToToken
+            .toFormat(props.token.decimals, { decimalSeparator })
+            .replace(new RegExp(`[${decimalSeparator}]?0+$`), '')
+        : ''
+
+    const parsedTokenAmount = parseInputAmount(convertedLocalToToken, decimalSeparator)
+
+    return {
+      token: {
+        amount: convertedLocalToToken,
+        bignum: parsedTokenAmount,
+        readable: getDisplayTokenAmount(parsedTokenAmount, props.token),
+      },
+      local: {
+        amount: parsedLocalAmount?.toFixed(2) ?? '',
+        bignum: parsedLocalAmount,
+        readable: getDisplayLocalAmount(parsedLocalAmount, localCurrencySymbol),
```

Comment:
@kathaypacific hmm, I was also thinking about that. The idea of the whole derived state is to have all the necessary computations happen only once per input change and for all the consumers to use those already computed values. From this perspective, `amount` is useful in the `handleToggleAmountType` callback which uses it for a corresponding amount type to set it as an input value whenever you switch between amount types. But yes, it can  be changed to `bignum.toString()` but that would probably negate the idea of a single computation cycle?

I do understand that this might feel like a premature optimization but considering that swap screen has a lot of processing on every rendering cycle and sometimes feels a bit laggy (at least that is usually my experience) I thought it would make sense to optimize calculations that are triggered on every input change to reduce amount of necessary computations whenever possible.

Reply:
@kathaypacific now as I'm thinking of it, considering that `amount` is used only to change the actual input value whenever you toggle between values there appears another problem: `bignum` is an un-rounded number, so `bignum.toString()` wouldn't be sufficient enough as we'll need to round/group/format that number which is more cumbersome.